packages/kbn-typed-react-router-config/src/use_params.ts

```diff
@@ -23,9 +23,9 @@ export function useParams(...args: any[]) {
     args.pop();
   }
 
-  const paths = args as string[];
-
-  const getParamsArgs = [...paths, location, optional] as [never, Location<any>, boolean];
-
-  return router.getParams(...getParamsArgs);
+  return useMemo(() => {
+    // @ts-expect-error
+    return router.getParams(...args, location, optional);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router, ...args, location, optional]);
```

Comment:
Would you mind including a broader comment on why `@ts-expect-error` and `eslint-disable-next-line react-hooks/exhaustive-deps` are needed here as It's not obvious at first glance?